mne/preprocessing/pca_obs.py

```diff
@@ -0,0 +1,314 @@
+"""Principle Component Analysis Optimal Basis Sets (PCA-OBS)."""
+
+# Authors: Emma Bailey <bailey@cbs.mpg.de>,
+#          Steinn Hauser Magnusson <hausersteinn@gmail.com>
+# License: BSD-3-Clause
+# Copyright the MNE-Python contributors.
+
+import math
+
+import numpy as np
+from scipy.interpolate import PchipInterpolator as pchip
+from scipy.signal import detrend
+from sklearn.decomposition import PCA
+
+from mne.io.fiff.raw import Raw
+from mne.utils import logger
+
+
+def apply_pca_obs(
```

Comment:
`verbose` does some nice logging magic
```suggestion
@verbose
def apply_pca_obs(
```

Reply:
Also added in f4082b4 :heavy_check_mark:. Thank you @larsoner !